src/Confluent.Kafka/Consumer.cs

```diff
@@ -194,18 +194,37 @@ public void Unsubscribe()
             => consumer.Unsubscribe();
 
         /// <summary>
-        ///     Update the assignment set to \p partitions.
+        ///     Update the assignment set to <param name="partitions" />.
         ///
-        ///     The assignment set is the set of partitions actually being consumed
-        ///     by the KafkaConsumer.
-        /// </summary>
-        /// <remarks>
-        ///     blocks
-        /// </remarks>
+        ///     The assignment set is the complete set of partitions to consume
+        ///     from.
+        /// </summary>
+        /// <param name="partitions">
+        ///     The set of partitions to consume from. If an offset value of
+        ///     -1001 (Offset.Invalid) is specified for a partition, consumption
```

Comment:
Nullable is not as elegant/simple:

1. We already have (and need to continue to use) some other mechanism for specifying special values. Using Nullable here introduces a separate mechanism.
2. It doubles memory requirements per offset.
3. It brings the GC into play.
4. There is no built in language construct in F# for Nullable.

The costs outweigh the benefit IMO.

Generally speaking, Nullable is loosing favor with me as something to be used in a public interface. Also, it's not used much in the .NET Framework BCLs and only really for peripheral things.